recipe-client-addon/lib/EventEmitter.jsm

```diff
@@ -20,15 +19,13 @@ this.EventEmitter = function(driver) {
       Promise.resolve()
         .then(() => {
           if (!(eventName in listeners)) {
-            driver.log(`EventEmitter: Event fired with no listeners: ${eventName}`);
+            log.info(`EventEmitter: Event fired with no listeners: ${eventName}`);
             return;
           }
-          // freeze event to prevent handlers from modifying it
-          const frozenEvent = Object.freeze(event);
```

Comment:
I probably haven't thought this through enough, but I thought that if heartbeat, which constructs the event emitter, clones it into the sandbox (and we throw away the original non-cloned reference, storing only the cloned version as `this.eventEmitter`) the instance that we call `emit` on has access to the object?